app/src/test/kotlin/org/eclipse/kuksa/testapp/model/ConnectionInfoTest.kt

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package org.eclipse.kuksa.testapp.model
+
+import io.kotest.core.spec.style.BehaviorSpec
+import io.kotest.matchers.shouldBe
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import java.io.File
+import java.io.FileWriter
+
+class ConnectionInfoTest : BehaviorSpec({
+    given("A custom ConnectionInfo with a custom Certificate") {
+        // Uri is an android specific class, which does not exist in plain JUnit. Methods like Uri.parse will simply
+        // return null due to the addition of testOptions.unitTests.isReturnDefaultValues true
```

Comment:
can only be fixed with Robolectric, but if I remember correctly there was an incompatibility of Kotest <-> Robolectric

Reply:
Exactly